### PR TITLE
K8s/Folders: Enhance k8s folder update

### PR DESCRIPTION
**Pull request by suntala**
**What is this feature?**

Include additional fields as part of folder update and include testing for the handler.

**Who is this feature for?**

Grafana Search and Storage

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/search-and-storage-team/issues/107

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/tests/apis/folder/folders_test.go [hunk 1]
```diff
@@ -386,6 +386,22 @@ func doFolderTests(t *testing.T, helper *apis.K8sTestHelper) *apis.K8sTestHelper
...
+			metav1.UpdateOptions{},
+		)
+		require.NoError(t, err)
+		spec, ok := updated.Object["spec"].(map[string]any)
+		require.True(t, ok)
+		title, ok := spec["title"].(string)
+		require.True(t, ok)
+		require.Equal(t, first.GetName(), updated.GetName())
+		require.Equal(t, first.GetUID(), updated.GetUID())
+		require.Equal(t, "Test folder (replaced from k8s; 1 item; PUT)", title)
```
Comment (filewalkwithme) [inline pkg/tests/apis/folder/folders_test.go:402]: Could we test the description field too?

Comment (suntala) [reply pkg/tests/apis/folder/folders_test.go:404]: Good idea. Added it here: [4022e4f](https://github.com/grafana/grafana/pull/95033/commits/4022e4fd844f76d99df2cb9aa2c8ebf436e1ce37)

**Review by filewalkwithme — APPROVED** [review]
:rocket: :rocket: :rocket:

File: pkg/tests/apis/folder/folders_test.go (hunk 1 above)
Comment (filewalkwithme) [reply pkg/tests/apis/folder/folders_test.go:404]: Thank you!

**Pull request merged**